src/Analyzers/CSharp/Analyzers/UseCollectionInitializer/CSharpUseCollectionInitializerAnalyzer.cs

```diff
@@ -16,11 +16,12 @@ internal sealed class CSharpUseCollectionInitializerAnalyzer : AbstractUseCollec
     MemberAccessExpressionSyntax,
     InvocationExpressionSyntax,
     ExpressionStatementSyntax,
-    ForEachStatementSyntax,
-    IfStatementSyntax,
```

Comment:
because the collection-initializer codepath doesn't need to handle Foreach/If-statements anymore, and because the helpers now do, all the generics around this were removed from the Use-Collection-Initializer inheritance hierarchy.